### PR TITLE
py-bpython: Remove Python 3.6 and add Python 3.10

### PR DESCRIPTION
**Pull request by judaew**
#### Description

See https://lists.macports.org/pipermail/macports-dev/2021-December/044043.html

- py-backports.cached-property: Remove python 3.6
- py-{curtsies,bpython}: Remove Python 3.6

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.1 21C52 x86_64
Xcode 13.2.1 13C100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by pmetzger** [comment]
LGTM. Arguably 3.10 might be added (where appropriate, if it passes tests).

**Comment by judaew** [comment]
@pmetzger Adding a new version of Python is like cancer, especially with tests. It takes a lot of time.

File: python/py-argh/Portfile [hunk 1]
```diff
@@ -21,11 +21,28 @@ checksums           rmd160  42da2aa1dfae654419acfc62db73d0fb27ecd23f \
                     sha256  e9535b8c84dc9571a48999094fda7f33e63c3f1b74f3e5f3ac0105a58405bb65 \
                     size    32913
 
-python.versions     27 35 36 37 38 39
+python.versions     27 35 36 37 38 39 310
```
Comment (pmetzger) [inline python/py-argh/Portfile:24]: Presumably we want to get rid of 35/36 here?

**Review by pmetzger — COMMENTED** [review]
I see what you mean about metastasis here. But it was good work!

File: python/py-argh/Portfile (hunk 1 above)
Comment (lpsinger) [reply python/py-argh/Portfile:24]: I agree. 36 is past end of life. If you remove this, then you can also remove the `if {${python.version} >= 37}` conditional.

**Review by lpsinger — CHANGES_REQUESTED** [review]
Shouldn't we also be dropping support for 2.7?

**Comment by lpsinger** [comment]
> > Shouldn't we also be dropping support for 2.7?
> 
> As I know, all package managers try to drop Python 2.7. But here the [digital_rf](https://ports.macports.org/port/digital_rf/details/) port depends on `py27-watchdog`, which in turn depends on `py27-argh`. I am not sure if this port can work with the Python 3.* version. Anyway, it is better to leave it to another PR.

According to the README file of digital_rf, it supports Python >= 3.5. I'd feel better about updating digital_rf first (yes, in a separate PR) so that the diff of the present PR is much smaller.

**Comment by pmetzger** [comment]
We can do this in two steps then, getting rid of the 2.7 support here after taking this initial step.

**Comment by judaew** [comment]
@pmetzger Can I merge it?

**Comment by pmetzger** [comment]
@judaew Go for it!

**Pull request merged**